projects/rocr-runtime/rocrtst/common/common.cc

```diff
@@ -405,6 +405,16 @@ hsa_status_t AcquirePoolInfo(hsa_amd_memory_pool_t pool,
   const size_t max_pool_size = 2*1024*1024*1024UL;
   pool_i->size = std::min(pool_i->size, max_pool_size);
 #endif
+  char *pool_size_limit_str = getenv("ROCRTST_LIMIT_POOL_SIZE");
+  if (pool_size_limit_str) {
+    char *end;
+    size_t pool_size_limit = strtoul(pool_size_limit_str, &end, 10);
+    if (pool_size_limit > pool_i->size) {
+      std::cout << "Warning: Pool size override > than reported size (override:"
+        << pool_size_limit << " reported:" << pool_i->size << ")" << std::endl;
+    }
+    pool_i->size = pool_size_limit;
```

Comment:
You will need pool_size_limit to be declared as a global variable and initialized to 0 (probably in common.h).

And then in the other section something similar to this:

float pool_size_limit_ratio = 1.0;
if (ag_type == HSA_DEVICE_TYPE_CPU) {
   pool_size_limit_ratio = pool_size_limit ? 0.9 : 0.7
}

uint64_t upper_bound = pool_size_limit_ratio * pool_sz;